### PR TITLE
Improve psql completions and add some related commands

### PR DESCRIPTION
**Pull request by janverb**
- Add completions for a few more postgresql commands
- Fix small issues in the existing `psql` completions, like stray commas
- Add missing `psql` options

### CODE REVIEW

File: share/completions/createdb.fish [hunk 1]
```diff
@@ -0,0 +1,21 @@
+complete -c createdb --no-files -a '(__fish_complete_pg_database)'
+
+# Options:
+complete -c createdb -s D -l tablespace -x -d "default tablespace for the database"
```
Comment (faho) [inline share/completions/createdb.fish]: Please capitalize the first letter for descriptions.

File: share/completions/pg_dump.fish [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+complete -c pg_dump -s '?' -l help -d "show this help, then exit"
+
+# Options controlling the output content:
+complete -c pg_dump -s a -l data-only -d "dump only the data, not the schema"
+complete -c pg_dump -s b -l blobs -d "include large objects in dump"
+complete -c pg_dump -s B -l no-blobs -d "exclude large objects in dump"
+complete -c pg_dump -s c -l clean -d "clean (drop) database objects before recreating"
+complete -c pg_dump -s C -l create -d "include commands to create database in dump"
+complete -c pg_dump -s E -l encoding -x -d "dump the data in encoding ENCODING"
+complete -c pg_dump -s n -l schema -x -d "dump the named schema(s) only"
```
Comment (faho) [inline share/completions/pg_dump.fish]: Just use the plural - "Dump the named schemas only".

People will get that they can also use one, and it's a bit shorter and easier to skim.

File: share/completions/pg_restore.fish [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# General options:
+complete -c pg_restore -s d -l dbname -x -a '(__fish_complete_pg_database)' -d "connect to database name"
+complete -c pg_restore -s f -l file -a '-' -r -d "output file name (- for stdout)"
+complete -c pg_restore -s F -l format -x -a "c d t" -d "backup file format (should be automatic)"
```
Comment (faho) [inline share/completions/pg_restore.fish]: From https://www.postgresql.org/docs/9.2/app-pgrestore.html:

```
c
custom

    The archive is in the custom format of pg_dump.
d
directory

    The archive is a directory archive.
t
tar

    The archive is a tar archive.
```

That information would be nice to have as a description:

```fish
complete -c pg_restore -s F -l format -x -a \
"c\t'Custom format' d\t'Directory archive' t\t'Tar archive'"
```

(and possibly also the long names)

Comment (faho) [reply share/completions/pg_restore.fish]: (same goes for pg_dump, of course)

Comment (janverb) [reply share/completions/pg_restore.fish]: Thank you. `pg_dump` and `pg_restore` seem to ignore everything but the first letter, so I left out the long names.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**